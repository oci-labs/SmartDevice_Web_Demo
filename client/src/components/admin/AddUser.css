.addUser {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-content: center; }
  .addUser .addUserModalButton {
    display: flex;
    width: 100%;
    margin: 22px 0px 0px;
    border: none;
    border-radius: 5px;
    color: white;
    font-size: 0.9em;
    height: 40px;
    justify-content: center; }
    .addUser .addUserModalButton.add {
      background-color: #fdaa18; }
