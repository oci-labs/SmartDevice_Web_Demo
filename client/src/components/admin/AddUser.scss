@import "../../custom.config.scss";

.addUser {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-content: center;
  .addUserModalButton {
    display: flex;
    width: 100%;
    margin: 22px 0px 0px;
    border: none;
    border-radius: 5px;
    color: white;
    font-size: 0.9em;
    height: 40px;
    justify-content: center;

    &.add {
      background-color: $primary-color;
    }
  }
}
