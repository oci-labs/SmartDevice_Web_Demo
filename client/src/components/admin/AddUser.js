import {connect} from 'react-redux';
import AddUser from './add-user';
import {actions} from 'react-redux-form';
import {addNewUser} from '../../actions';
import {toggleUserModal} from '../../redux-modules/view/actions';
import {selectShowUserModal} from '../../selectors/view-selectors';
import {selectFormIsValid, selectNewUser} from '../../selectors/user-selectors';

const mapStateToProps = state => ({
  showModal: selectShowUserModal(state),
  isValid: selectFormIsValid(state)
});

const mapDispatchToProps = dispatch => ({
  onAddUser: () =>
    dispatch((dispatch, getState) => {
      if (selectFormIsValid(getState())) {
        dispatch(addNewUser(selectNewUser(getState())));
        dispatch(toggleUserModal());
        dispatch(actions.reset('forms.addUser'));
      } else {
        console.log('Add user form is not valid.');
      }
    }),
  onToggleModal: () => {
    dispatch(actions.reset('forms.addUser'));
    dispatch(toggleUserModal());
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(AddUser);
