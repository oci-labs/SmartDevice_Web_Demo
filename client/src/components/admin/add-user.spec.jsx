/* global describe, it, expect, jest, renderedSnapshot, shallow, mount */
import React from 'react';
import { Modal, ModalFooter } from 'reactstrap';
import AddUser from './add-user';
import { Provider } from 'react-redux';
import { ReactWrapper } from 'enzyme';

describe('AddUser component', () => {
  it('should render as expected', () => {
    renderedSnapshot(<AddUser showModal={false} onAddUser={jest.fn()} onToggleModal={jest.fn()} />);
  });

  it('should pass appropriate values to modal', () => {
    const onToggle = jest.fn();
    const rendered = shallow(
      <AddUser showModal isValid onAddUser={jest.fn()} onToggleModal={onToggle} />
    );

    const modal = rendered.find(Modal);
    expect(modal.prop('isOpen')).toBeTruthy();
    expect(modal.prop('toggle')).toBe(onToggle);
  });

  it('should enable the add button if the form is valid', () => {
    const rendered = shallow(
      <AddUser showModal isValid onAddUser={jest.fn()} onToggleModal={jest.fn()} />
    );

    const addButton = rendered.find('button.addUserModalButton');
    expect(addButton.prop('disabled')).toBeFalsy();
    expect(addButton.prop('className')).toEqual(expect.stringContaining('enabled'));
  });

  it('should disable the add button if the form is not valid', () => {
    const rendered = shallow(
      <AddUser showModal isValid={false} onAddUser={jest.fn()} onToggleModal={jest.fn()} />
    );

    const addButton = rendered.find('button.addUserModalButton');
    expect(addButton.prop('disabled')).toBeTruthy();
    expect(addButton.prop('className')).not.toEqual(expect.stringContaining('enabled'));
  });

  it('should apply onAddUser to the add button onClick', () => {
    const addUser = jest.fn();
    const rendered = shallow(
      <AddUser showModal isValid={false} onAddUser={addUser} onToggleModal={jest.fn()} />
    );

    const addButton = rendered.find('button.addUserModalButton');
    expect(addButton.prop('onClick')).toBe(addUser);
  });
});
