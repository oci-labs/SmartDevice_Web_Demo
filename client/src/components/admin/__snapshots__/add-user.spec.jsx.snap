// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUser component should render as expected 1`] = `
<div
  class="addUser"
>
  <div>
    <button
      class="addUserButton"
    >
      Add a User
    </button>
  </div>
</div>
`;
